core/input/mouse.js

```diff
@@ -108,27 +116,74 @@ Mouse.prototype = {
         this._handleMouseButton(e, 0);
     },
 
+    // Mouse wheel events are sent in steps over VNC. This means that the VNC
+    // protocol can't handle a wheel event with specific distance or speed.
+    // Therefor, if we get a lot of small mouse wheel events we combine them.
+    _generateWheelStepX: function (pos) {
+
+        if (this._accumulatedWheelDeltaX < 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 5);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 5);
+        } else if (this._accumulatedWheelDeltaX > 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 6);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 6);
+        }
+
+        this._accumulatedWheelDeltaX = 0;
+    },
+
+    _generateWheelStepY: function (pos) {
+
+        if (this._accumulatedWheelDeltaY < 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 3);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 3);
+        } else if (this._accumulatedWheelDeltaY > 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 4);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 4);
+        }
+
+        this._accumulatedWheelDeltaY = 0;
+    },
+
     _handleMouseWheel: function (e) {
-        if (!this._focused) { return; }
+        if (!this._focused || !this._onMouseButton) { return; }
+
+        window.clearTimeout(this._wheelStepXTimer);
+        window.clearTimeout(this._wheelStepYTimer);
```

Comment:
Don't you need to clear the variables as well in order to avoid the risk of stopping unrelated timers the next round?

Reply:
I tested it quite extensively and even printing timestamps, it seemed to behave properly. But what do you mean with unrelated timers?

Reply:
Timer ids are simple integers, so it is possible for re-use. So if you get id 45 and then clear it, some other code might register a new timeout and also get 45 since that id is now free.

Reply:
Oh, I thought the id was a longer almost unique identifier. I'll just null the variables after clearing the timeout then.
Seems like null-checks are not needed since the functions don't error when called with null.